src/org/odk/collect/android/provider/FormsProvider.java

```diff
@@ -695,6 +693,19 @@ public int update(Uri uri, ContentValues values, String where,
 		return count;
 	}
 
+    @NonNull
+    private String[] prepareWhereArgs(String[] whereArgs, String formId) {
+        String[] newWhereArgs;
+        if (whereArgs == null || whereArgs.length == 0) {
+            newWhereArgs = new String[] {formId};
+        } else {
+            newWhereArgs = new String[(whereArgs.length + 1)];
+            newWhereArgs[0] = formId;
+            System.arraycopy(whereArgs, 0, newWhereArgs, 1, whereArgs.length);
+        }
+        return newWhereArgs;
+    }
+
 	static {
```

Comment:
Maybe extract this method to be reused by the InstanceProvider too?

Reply:
I started doing that and then decided to keep our implementation as close as possible to the odk-collect version from upstream. This will help us take any future patches.